src/toolbar/ToolbarButton.jsx

```diff
@@ -0,0 +1,77 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { adjustColor } from './utility';
+
+const classNames = require('classnames');
+
+const ButtonIcon = (props) => {
+  const {
+    img_url: imgUrl, icon, color, enabled,
+  } = props;
+
+  if (icon) {
+    return <i className={icon} style={{ color: adjustColor(color, enabled) }} />;
+  }
+
+  if (imgUrl && !icon) {
+    return <img alt="Button icon" src={imgUrl} data-enabled={imgUrl} data-disabled={imgUrl} />;
+  }
+
+  return null;
+};
+
+ButtonIcon.propTypes = {
+  img_url: PropTypes.string,
+  icon: PropTypes.string,
+  color: PropTypes.string,
+  enabled: PropTypes.bool,
+};
+
+/* FIXME
+ * ng-class="{active: props.selected, disabled: !props.enabled}"
+ * ng-hide="props.hidden"
+ * FIXME: where did this come from   style={props.enabled && props.hidden ? {} : { display: 'none !important' }}
+ * style={props.hidden ? { display: 'none !important' } : {}}
+ */
+export const ToolbarButton = props => (
+  <button
+    type="button"
+    id={props.id}
+    name={props.name}
+    title={props.title}
+    className={classNames('btn btn-default', { active: props.selected, disabled: !props.enabled })}
+    data-explorer={props.explorer}
```

Comment:
could you get rid of these explicit keys in favor of generic data?

No part of toolbar code (except for `miqToolbarOnClick`) should be aware that data-function and data-function-data and data-click even exist.. (and all the others).

Reply:
I guess yes. Just I'd love to get a functional parity with the current implementation first and then improve.

But right, using generic `data` should be fine as long as we do not handle some `data-` key in some special way in the current implementation.

Reply:
I will be removing all the `data-` tags because the `onClick` handler now uses `props` directly. So this will go away.